vsbot :
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Velocity Controllers ------------------------------------
  joint1_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: joint1
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint2_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: joint2
    pid: {p: 100.0, i: 0.01, d: 10.0}
  
  # Position Controllers ------------------------------------
  joint1_position_controller:
    type: velocity_controllers/JointPositionController
    joint: joint1
    pid: {p: 1.0, i: 0.00, d: 0.1}

  joint2_position_controller:
    type: velocity_controllers/JointPositionController
    joint: joint2
    pid: {p: 1.0, i: 0.00, d: 0.1}

# Jacobian Estimation Parameters ------------------------------------
  estimation:
    epsilon: 1.0e-5 # update threshold or convergence condition
    gamma: 10.0e-5 # learning rate
    gamma2: 10.0e-7 # learning rate during servoing
    window: 100 # window size
    rate: 30 # control loop rate
  control:
    thresh: 0.01 # visual servoing convergence threshold
    goal_pos_x: 250
    goal_pos_y: 145
    window: 25 # window size
    lam: 5.0e-3 # servoing gain

# Goal Position for End Effector ------------------------------------

  vs_baseline:
    
     goal_position_x: 96 # x co-ordinates 
     goal_position_y: 166 # Y co-ordinates
     pub_rate: 30 # image Publisher rate


  


